src/web_app/js/peerconnectionclient.js

```diff
@@ -360,5 +375,215 @@ PeerConnectionClient.prototype.onRemoteStreamAdded_ = function(event) {
+    getUserMedia: 'getUserMedia',
+    createOffer: 'createOffer',
+    createAnswer: 'createAnswer',
+    setLocalDescription: 'setLocalDescription',
+    setRemoteDescription: 'setRemoteDescription',
+    addIceCandidate: 'addIceCandidate'
+  };
+
+  // Only report supported error types to the callstats backend.
+  if (supportedWebrtcFuncNames[funcName]) {
+    // Some error objects (gUM) have meaningful info in the name
+    // property/getter.
+    var filteredError = (funcName === 'getUserMedia' ? error.name : error);
+    this.callstats.reportError(this.pc_, this.conferenceId,
+      supportedWebrtcFuncNames[funcName], new DOMException(filteredError),
+      localSdp, remoteSdp);
+  }
+};
+
+PeerConnectionClient.prototype.initCallstats_ = function() {
+  trace('Init callstats.');
+  // jscs:disable requireCapitalizedConstructors
+  /* jshint newcap: false */
+  this.callstats = new callstats(null, io, jsSHA);
+  // jscs:enable requireCapitalizedConstructors
+  /* jshint newcap: true */
+
+  var appId = this.params_.callstatsParams.appId;
+  var appSecret = this.params_.callstatsParams.appSecret;
+  this.userId = this.params_.roomId + (this.isInitiator_ ? '-0' : '-1');
+  var statsCallback = null;
+  var configParams = null;
+  var callback = function(status, msg) {
+    trace('Init status: ' + status + ' msg: ' + msg);
+  };
+  this.callstats.initialize(appId, appSecret, this.userId, callback,
+      statsCallback, configParams);
+};
+
+// Setup the callstats api and attach it to the peerconnection.
+PeerConnectionClient.prototype.setupCallstats_ = function() {
+  // Check dependencies.
+  if (typeof io !== 'function' && typeof jsSHA !== 'function')  {
+    trace('Callstats dependencies missing, stats will not be setup.');
+    return;
+  }
+
+  // Need to catch the error otherwise the peerConnection creation
+  // will fail.
+  try {
+    // Authenticate with the callstats backend.
+    this.initCallstats_();
+    this.callStatsAttachedToPc = false;
+
+    trace('Set up callstats.');
+    this.conferenceId = this.params_.roomId;
+    this.remoteUserId = this.params_.roomId + (this.isInitiator_ ? '-1' : '-0');
+    // Multiplex should be used when sending audio and video on a peerConnection.
+    // http://www.callstats.io/api/#enumeration-of-fabricusage
+    // TODO: Might need to change this dynamically if an audio/video only call.
+    var usage = this.callstats.fabricUsage.multiplex;
+    var callback = function(status, msg) {
+      trace('Callstats status: ' + status + ' msg: ' + msg);
+    }.bind(this);
+    // Hookup the callstats api to the peerConnection object.
+    this.callstats.addNewFabric(this.pc_, this.remoteUserId, usage,
+        this.conferenceId, callback);
+  } catch (error) {
+    trace('Callstats could not be set up: ' + error);
+  }
+};
+
+// Associate send SSRC's with media stream tracks and user ID in the
+// callstats backend.
+PeerConnectionClient.prototype.bindSendMstToIdForCallstats_ = function() {
```

Comment:
This seems like it could be unified with the function below. Also, I don't understand the comment.